examples/src/examples/gizmos/transform-rotate.example.mjs

```diff
@@ -71,6 +73,7 @@ camera.addComponent('camera', {
     farClip: 1000
 });
 camera.addComponent('script');
+camera.script.create(InfiniteGrid);
```

Comment:
Wait, why would the grid be tagged on the camera?

Reply:
Oh it needs the camera for the view projection matrix and for switching between perspective and orthographic modes

Reply:
@mvaligursky or @slimbuck might be able to advise here but I would have expected the implementation to maybe add the grid to a new layer and in a callback before rendering the layer, get the current camera info. I'm all for making script usage easy and guessable...and I would not have guessed that a grid would need to be tagged on an entity that also has a camera component.

Reply:
Yea I think the API for it specifically needs to be adjusted for sure.